test/support/component_helpers.rb

```diff
@@ -7,54 +7,38 @@ def assert_has_component_title(title, context = nil)
     end
   end
 
-  def assert_has_button_component(text, attrs = {})
-    all(shared_component_selector('button')).each do |button|
-      data = JSON.parse(button.text).symbolize_keys
-      if attrs[:href] && data[:href]
-        next unless text == data.delete(:text) && attrs[:href] == data[:href]
-      else
-        next unless text == data.delete(:text)
-      end
-      data.delete(:data_attributes) if data[:data_attributes].blank?
-      return assert_equal attrs, data
-    end
+  def assert_has_button_component(text)
+    assert page.has_css?("button", text: text)
+  end
 
-    fail_button_not_found(text)
+  def assert_has_button_component_as_link(text, attrs = {})
+    assert page.has_css?(process_button_attributes(attrs), text: text)
   end
 
   def refute_has_button_component(text, attrs = {})
-    all(shared_component_selector('button')).each do |button|
-      data = JSON.parse(button.text).symbolize_keys
-      if attrs[:href] && data[:href]
-        next unless text == data.delete(:text) && attrs[:href] == data[:href]
-      else
-        next unless text == data.delete(:text)
-      end
-      data.delete(:data_attributes) if data[:data_attributes].blank?
-      return refute_equal attrs, data
-    end
+    assert page.has_no_css?(process_button_attributes(attrs), text: text)
   end
 
   def click_button_component(text)
```

Comment:
I think instead of changing the functionality here we can search/replace `click_button_component` with `click_on`?

It's used here already: https://github.com/alphagov/frontend/blob/8c115a328797ff3b8b81912f1b14dec73609f2f8/test/integration/licence_test.rb#L550

Reply:
Great, will change.